src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -399,8 +400,11 @@ bool ConnectivityManagerImpl::_IsWiFiStationConnected()
     bool ret            = false;
     const gchar * state = nullptr;
 
+    mWpaSupplicantLock.lock();
```

Comment:
How about using the stl RAII style lock guard, rather than manually lock/unlock ? https://en.cppreference.com/w/cpp/thread/lock_guard